scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+      this.extractedCopyLink = ExtractedCopyLink.newBuilder().apply {
+        url = "https://local-copy/bsd-license"
+      }.build()
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val depsThatNeedInterventionSet =
+      MavenDependenciesRetriever.getDependenciesThatNeedIntervention(mavenDependenciesList)
+    assertThat(depsThatNeedInterventionSet).isEmpty()
+  }
+
+  @Test
+  fun testGetDepsThatNeedIntervention_depsDetailsIncomplete_returnsCorrectDepsSet() {
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder().apply {
+        url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      }.build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Fabric Software and Services Agreement"
+      this.originalLink = "https://fabric.io/terms"
+      this.isOriginalLinkInvalid = true
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_INVALID_LINKS
+        this.artifactVersion = IO_FABRIC_VERSION
+        this.addLicense(license2)
+      }.build()
+    )
+
+    val depsThatNeedInterventionSet =
+      MavenDependenciesRetriever.getDependenciesThatNeedIntervention(mavenDependenciesList)
+    assertThat(depsThatNeedInterventionSet.size).isEqualTo(2)
+    assertIsDependency(
+      dependency = depsThatNeedInterventionSet.elementAt(0),
+      artifactName = DEP_WITH_NO_LICENSE,
+      artifactVersion = PROTO_LITE_VERSION
+    )
+    assertIsDependency(
+      dependency = depsThatNeedInterventionSet.elementAt(1),
+      artifactName = DEP_WITH_INVALID_LINKS,
+      artifactVersion = IO_FABRIC_VERSION
+    )
+  }
+
+  @Test
+  fun testRetrieveMavenDepList_emptyPbFile_returnsEmptyDepsList() {
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val mavenDependencyList = MavenDependencyList.newBuilder().build()
+
+    mavenDependencyList.writeTo(pbFile.outputStream())
```

Comment:
Ditto here: need to use ``use{}`` to ensure the file stream gets closed.

Reply:
Done.